src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -652,31 +674,32 @@ public static ConfigCatClient get(final String sdkKey) {
     }
 
     /**
-     * Get a singleton ConfigCat Client for the sdk key with the options, create a new client if not existed yet.
-     * If the client already exist the options are ignored.
+     * Creates a new or gets an already existing ConfigCatClient for the given sdkKey.
      *
-     * @param sdkKey  the client sdk key.
-     * @param options the client initializer options.
-     * @return a singleton client.
+     * @param sdkKey          the SDK Key for to communicate with the ConfigCat services.
+     * @param optionsCallback the options callback to configure the created ConfigCatClient instance.
+     * @return the ConfigCatClient instance.
      */
-    public static ConfigCatClient get(final String sdkKey, final Options options) {
+    public static ConfigCatClient get(String sdkKey, Consumer<Options> optionsCallback) {
         if (sdkKey == null || sdkKey.isEmpty()) {
             throw new IllegalArgumentException("'sdkKey' cannot be null or empty.");
         }
 
         synchronized (INSTANCES) {
-            Options clientOptions = options;
+            Options clientOptions = new Options();
 
             ConfigCatClient client = INSTANCES.get(sdkKey);
             if (client != null) {
-                if (clientOptions != null) {
+                if (optionsCallback != null) {
                     client.logger.warn("Client for '" + sdkKey + "' is already created and will be reused; options passed are being ignored.");
```

Comment:
the passed options are ignored because the client ...

Reply:
They should be ignored if we already have a client. We do not override the options after the client is created, so in this case if we have a client the options passed again we warn the user the options will be not used.

Reply:
My comment meant to be the new message.